webapp/src/package.ts

```diff
@@ -1000,4 +1000,53 @@ export function getExtensionOfFileName(filename: string) {
 
 async function getHostCacheAsync(): Promise<pxt.BrowserUtils.IDBObjectStoreWrapper<{id: string, val: string}>> {
     return getObjectStoreAsync(HOSTCACHE_TABLE)
+}
+
+export function getProjectToolboxFilters() {
+    const filters: pxt.editor.ProjectFilters = {};
+
+    const deps = mainPkg.sortedDeps();
+    // sort the dependencies by level and then id. lower level overrides
+    // higher level
+    deps.sort((a, b) => {
+        if (a.level === b.level) {
+            return pxt.U.strcmp(a.id, b.id);
+        }
+        return b.level - a.level;
+    });
+
+    const applyProjectFilter = (projectFilter: pxt.PackageConfig["toolboxFilter"], key: keyof pxt.PackageConfig["toolboxFilter"]) => {
```

Comment:
I just wanted to ask about this because I don't think I've seen `keyof` being used before. I looked at the documentation, but just wanted some clarification. 

So using `keyof` here is basically saying, "hey, the type we are expecting has to be the keys defined within this object. If the string (in our case) or type does not exist within this object, we can't accept that parameter". And the benefit of using something like this is that we can use the object itself as a growing type definition if we ever have any fields that we want to add to the `toolboxFilter` object rather than having to create a separate type definition.

Reply:
yup!